keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,416 @@
+
+### User Stories [optional]
+
+#### Story 1
+As a cluster admin, I would like my node to be as performant as possible, and not waste resources by duplicating stats collection by CRI and cAdvisor.
+
+#### Story 2
+As a node maintainer, I would like to clarify the sources of truth for each stat the Kubelet reports to the Summary API, so that I can better find which component is to investigate for any issues.
+
+### A note on history
+There have been conversations in the past on [deprecating the stats Summary API][#core-proposal], in favor of allowing an on-demand model of stats gathering/reporting.
+This KEP does not attempt to implement that. These stats are largely considered [a part of Kubernetes][#keep-summary]. The stated goal of this KEP is to reduce the overhead
+on two entities reporting metrics, not totally changing what stats the Kubelet reports (with the exception of Accelerator and UserDefinedMetrics).
+
+Thus, this KEP largely the plan described [here][#plan], with some changes:
+
+- the `/metrics/cadvisor` endpoint will likely not be exposed
+	- **TODO** will this break users?
+- CRI is used for all of the monitoring endpoints related to Containers and Pods (except Volume and Ephemeral Storage)
+- CRI is used to provide metrics for eviction.
+	- **TODO**: This is possible with just the stats Summary API, right?
+
+
+[core-proposal]: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/core-metrics-pipeline.md#proposed-core-metrics
+[keep-summary]: https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-666636130
+[plan]: https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-724928827
+### Risks and Mitigations
+
+- To properly move to CRI stats, it is likely there are some metrics that we'll want to not support (Accelerator/UserDefined). We should be careful to not break entities that rely on these metrics.
+- A large part of this work is changing the source of the Summary API metrics. In doing so, there is a risk that collecting from a new source will change how the stats look in aggregate (and risk bugs popping up in new areas).
+- cAdvisor has a long history of collecting these stats. There is a risk that changing the source of the stats to the CRI implementation can cause performance regressions
+as cAdvisor is fine tuned to perform in an adequate manner.
+    - CRI implementations should do performance regression analyses to ensure the change does not regress too much.
+
+## Graduation Criteria
+
+
+## Implementation History
+
+
+## Drawbacks [optional]
+
+CRI runtimes will each have to implement additional interface to support full stats, rather than all metric collection being unified by cAdvisor. 
+
+Note: This is by design as this will enable to decouple runtime implementation details further from kubelet.
+
+## Alternatives [optional]
```

Comment:
Other alternatives that come to mind, or have been proposed previously:

* Have cAdvisor implement the summary API.  A cAdvisor daemonset could be a drop-in replacement for the summary API.
* Don't keep supporting the summary API.  Replace it with a "better" format, like prometheus.  Or help users migrate to equivalent APIs that container runtimes already expose for monitoring.

I'm not saying we should necessarily do either of those, but they are possible alternatives.

Reply:
added them